test/framework/toolchain.py

```diff
@@ -955,6 +955,49 @@ def test_precision_flags(self):
 
                 self.modtool.purge()
 
+    def test_search_path_cpp_headers(self):
+        """Test functionality behind search-path-cpp-headers option"""
+        cpp_headers_mode = {
+            "CPPFLAGS": ["CPPFLAGS"],
+            "CPATH": ["CPATH"],
+            "INCLUDE_PATHS": ["C_INCLUDE_PATH", "CPLUS_INCLUDE_PATH", "OBJC_INCLUDE_PATH"],
+        }
+        # test without toolchain option
+        for build_opt in cpp_headers_mode:
+            init_config(build_options={"search_path_cpp_headers": build_opt, "silent": True})
+            tc = self.get_toolchain("foss", version="2018a")
+            with self.mocked_stdout_stderr():
+                tc.prepare()
+                for env_var in cpp_headers_mode[build_opt]:
+                    assert_fail_msg = (
+                        f"Variable {env_var} required by search-path-cpp-headers build option '{build_opt}' "
+                        "not found in toolchain environment"
+                    )
+                    self.assertIn(env_var, [*tc.variables], assert_fail_msg)
+                self.modtool.purge()
+        # test with toolchain option
+        for build_opt in cpp_headers_mode:
+            init_config(build_options={"search_path_cpp_headers": build_opt, "silent": True})
+            for tc_opt in cpp_headers_mode:
+                tc = self.get_toolchain("foss", version="2018a")
+                tc.set_options({"search-path-cpp-headers": tc_opt})
+                with self.mocked_stdout_stderr():
+                    tc.prepare()
+                    for env_var in cpp_headers_mode[tc_opt]:
+                        assert_fail_msg = (
+                            f"Variable {env_var} required by search-path-cpp-headers toolchain option '{tc_opt}' "
+                            "not found in toolchain environment"
+                        )
+                        self.assertIn(env_var, [*tc.variables], assert_fail_msg)
+                self.modtool.purge()
+        # test wrong toolchain option
+        tc = self.get_toolchain("foss", version="2018a")
+        tc.set_options({"search-path-cpp-headers": "WRONG_MODE"})
+        with self.mocked_stdout_stderr():
+            error_pattern = "Unknown value selected for option search-path-cpp-headers"
+            self.assertErrorRegex(EasyBuildError, error_pattern, tc.prepare)
+        self.modtool.purge()
```

Comment:
Do we need the purge at the end of the test?

Reply:
We do it in all other toolchain tests. I think it does not hurt to clean up after the test.

Reply:
Just looks odd to me.

My understanding is that the purge only resets the environment which we restore already in the teardown of the base testcase class. If we really want to do that at the end of all those testcases we could move it to the `teardown` of this test-class: https://github.com/easybuilders/easybuild-framework/blob/8cbb705f81f3aa9fe7e5adbee1e0e486e575f7e8/test/framework/toolchain.py#L77-L83

Reply:
I do not disagree with you, but this is out of scope for this PR. I will not change the workflow of tests here.